### PR TITLE
MachXO2 improvements

### PR DESCRIPTION
**Pull request by mmicko**
This one requires https://github.com/YosysHQ/prjtrellis/pull/211 and https://github.com/YosysHQ/prjtrellis-db/pull/5 to be applied.

- Adds use of "-device" instead of handling too many variations (like nexus)
- Extends chipdb to support device selection.
- Adds basic display of BELs in GUI mode
- Makes implementation easier to extend for new models (MachXO in future)

### CODE REVIEW

**Comment by whitequark** [comment]
@mmicko Do you think I should ship MachXO2 in YoWASP?

**Comment by mmicko** [comment]
@whitequark not for now I think. I do it as part of oss-cad-suite, but for now only semi usable chip is 1200HC

**Comment by whitequark** [comment]
@mmicko Got it. Feel free to ping me anytime once it's more usable and I'll ship it.

File: machxo2/arch.cc [hunk 1]
```diff
@@ -21,12 +21,14 @@
...
 #include <math.h>
 #include "embed.h"
+#include "gfx.h"
 #include "nextpnr.h"
 #include "placer1.h"
 #include "placer_heap.h"
 #include "router1.h"
 #include "router2.h"
 #include "util.h"
+#include "chipdb/available.h"
```
[the remark this answers is not in the document]
Comment (cr1901) [reply machxo2/arch.cc:31]: Fixed with the following patch (I don't do in-tree builds):

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f48a894c..fac612ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -332,7 +332,7 @@ foreach (family ${ARCH})
         endforeach()

         # Include family-specific source files to all family targets and set defines appropriately
-        target_include_directories(${target} PRIVATE ${family}/ ${CMAKE_CURRENT_BINARY_DIR}/generated/)
+        target_include_directories(${target} PRIVATE ${family}/ ${CMAKE_CURRENT_BINARY_DIR}/generated/ ${CMAKE_CURRENT_BINARY_DIR}/${family}/)
         target_compile_definitions(${target} PRIVATE NEXTPNR_NAMESPACE=nextpnr_${family} ARCH_${ufamily} ARCHNAME=${family})
         target_link_libraries(${target} LINK_PUBLIC ${Boost_LIBRARIES} ${link_param})
         if (NOT MSVC)
```

**Review by cr1901 — APPROVED** [review]
The test bitstreams I've used in the past still work with these changes, so good to go on my end.

File: machxo2/arch.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmicko) [reply machxo2/arch.cc]: Thanks, I have placed it now in generated, it makes more sense to be there, but renamed so we can use similar approach in other architectures in future as well without conflict.

**Comment by gatecat** [comment]
I think the final step in this series is to update `TRELLIS_REVISION` in [arch_ci.yml](https://github.com/YosysHQ/nextpnr/blob/master/.github/workflows/arch_ci.yml) with the now-merged trellis head and then make sure xo2 ci passes

**Pull request merged**